app/src/main/java/com/realityexpander/tasky/agenda_feature/domain/AgendaItem.kt

```diff
@@ -1,32 +1,151 @@
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
 import java.time.LocalDateTime
+import java.util.*
 
 // NOTE: Skeleton data structures for the Agenda feature
 // **temporary** for UI development
 
+@OptIn(ExperimentalSerializationApi::class)
+@Serializer(forClass = LocalDateTime::class)
+object LocalDateTimeSerializer : KSerializer<LocalDateTime> {
+    override val descriptor = PrimitiveSerialDescriptor("LocalDateTime", PrimitiveKind.STRING)
+
+    override fun deserialize(decoder: Decoder): LocalDateTime {
+        return LocalDateTime.parse((decoder.decodeString()))
+    }
+
+    override fun serialize(encoder: Encoder, value: LocalDateTime) {
+        encoder.encodeString(value.toString())
+    }
+}
+
 abstract class AgendaItem(
-    val id: String,
-    val title: String,
-    val description: String,
-    val remindAt: LocalDateTime,
+    open val id: UuidStr,
+    open val title: String,
+    open val description: String? = null,
+    @Serializable(with = LocalDateTimeSerializer::class)
+    open val remindAt: LocalDateTime,
 ) {
     abstract fun isEmpty(): Boolean
 
+    @Parcelize
     class Event(
-        id: String,
-        title: String,
-        description: String,
+        override val id: UuidStr,
+        override val title: String,
+        override val description: String? = null,
+        @Serializable(with = LocalDateTimeSerializer::class)
+        override val remindAt: LocalDateTime,
+        @Serializable(with = LocalDateTimeSerializer::class)
         val from: LocalDateTime,
+        @Serializable(with = LocalDateTimeSerializer::class)
         val to: LocalDateTime,
-        remindAt: LocalDateTime,
-        val attendeeIds: List<String>? = null, // todo: use UUID instead of string
-    ): AgendaItem(id, title, description, remindAt) {
+        val attendeeIds: List<UuidStr>? = null,
+        val photos: List<UuidStr>? = null,
+    ): AgendaItem(id, title, description, remindAt), Parcelable {
+
         companion object {
             val EMPTY = Event("", "", "", LocalDateTime.MIN, LocalDateTime.MIN, LocalDateTime.MIN, emptyList())
         }
 
         override fun isEmpty() = this == EMPTY
+
+        fun copy(
+            id: UuidStr = this.id,
+            title: String = this.title,
+            description: String? = this.description,
+            remindAt: LocalDateTime = this.remindAt,
+            from: LocalDateTime = this.from,
+            to: LocalDateTime = this.to,
+            attendeeIds: List<UuidStr>? = this.attendeeIds,
+            photos: List<UuidStr>? = this.photos,
+        ) = Event(
+            id = id,
+            title = title,
+            description = description,
+            remindAt = remindAt,
+            from = from,
+            to = to,
+            attendeeIds = attendeeIds,
+            photos = photos,
+        )
```

Comment:
I'd actually prefer to just make `Event`, `Task` and `Reminder` all data classes here, so you don't need to create copy, equals and hashCode functions for them which are all really helpful. You just need to use different field names for them than for the `AgendaItem` constructor

Reply:
OK, I will separate them out.